jcl/src/java.base/share/classes/java/lang/StringBuilder.java

```diff
@@ -2261,7 +2261,43 @@ public StringBuilder append(CharSequence sequence, int start, int end) {
 	if (sequence == null) {
 		return append(String.valueOf(sequence), start, end);
 	} else if (sequence instanceof String) {
-		return append(((String)sequence).substring(start, end));
+		if (start >= 0 && end >= 0 && start <= end && end <= ((String) sequence).lengthInternal()) {
```

Comment:
Can `String sequenceString = (String) sequence;` be declared before the `if` so the redundant cast can be avoided?